assignments/lambda-classes.js

```diff
@@ -1 +1,116 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor ({name, age, location}) {
+        this.name = name
+        this.age = age
+        this.location = location
+    }
+    speak () {
+        return `Hello my name is ${this.name}, I am from ${this.location}`
+    }
+}
+
+//having a bit of fun switching constructor and super around
+class Instructor extends Person {
+    constructor (attr) {
+        super (attr);
+        this.specialty = attr.specialty;
+        this.favLanguage = attr.favLanguage;
+        this.catchPhrase = attr.catchPhrase
+    }
+
+    demo(subject) {
+        return `Today we are learning about ${subject}`
+}
+    grade(grade, student, subject) { 
+        return `${student} receives a perfect score in ${subject}`
+    }
+    finalGrade(student){
+        return `${student}'s Final grade is ${Math.random() * 100}`
+    }
+   
+}
+
+class Student extends Person {
+    constructor({previousBackground, className, favSubjects, grade = 50}){
+        super(...arguments);
+        this.previousBackground = previousBackground
+        this.className = className;
+        this.favSubjects = favSubjects
+    }
+    listsSubjects(){
+        return `${this.favSubjects}`
+    }
+
+    PRAssignment(subject){
+        return `${this.name} has submitted a PR for ${subject}`
+    }
+    sprintChallenge(){
+        return `${this.name} has begun sprint challenge on ${this.subject}`
+    }
+    graduation(){
+        if (Instructor.finalGrade > 70){
+            return `Pass`
+        }
+        else {
+            return `Fail`
+        }
+    }
+}
+
+// some more fun
+class ProjectManager extends Person {
+    constructor (attr) {
+    super (attr);
+    this.gradClassName = attr.gradClassName;
+    this.favInstructor = attr.favInstructor;
+    }
+    standUp(channel){
+        return `${this.name} announces to ${channel}, @channel 
+        standy times`
+    }
+    debugsCode (name, subject) {
+        return `${this.name} debugs ${name}'s code on ${subject}`
+    }
+    finalGrade(student){
+        return `${student}'s Final grade is ${Math.random() * 100}`
+    }
+}
+
+const testP = new Person ({name: "Fred", age: 30, 
+location: "Outer space"})
+
+console.log(testP);
+console.log(testP.speak())
+
+
+const testI = new Instructor ({name: "Jim", specialty: "Nothing", favLanguage: "English", age: 45, location: "California", catchPhrase: "It's not my fault the kids are failing"})
+
+console.log(testI)
+console.log(testI.name)
```

Comment:
Great job testing each of the constructors! it really highlights the separation and relationship of each object.